src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -433,26 +432,29 @@ case class JesBackend(actorSystem: ActorSystem)
       * Commands in WDL tasks can also generate input files.  For example: ./my_exec --file=${write_lines(arr)}
       *
       * write_lines(arr) would produce a string-ified version of the array stored as a GCS path.  The next block of code
-      * will use the file evaluator on each expression in the command to write these files and return back paths to
-      * those files so JES can localize them
+      * will go through each ${...} expression within the task's command section and find all write_*() ASTs and
+      * evaluate them so the files are written to GCS and the they can be included as inputs to Google's Pipeline object
       */
     val commandExpressions = backendCall.call.task.commandTemplate.collect({
       case x: ParameterCommandPart => x.expression
     })
 
-    val commandFiles = commandExpressions.map({ e =>
-      val k = e.toWdlString.md5SumShort
-      val v = e.evaluateFiles(backendCall.lookupFunction(Map.empty), backendCall.engineFunctions, WdlAnyType) match {
-        case Success(files) => files
-        case Failure(ex) => Seq.empty[WdlFile]
-      }
-      k -> v
-    }).toMap
+    val writeFunctionAsts = commandExpressions.map(_.ast).flatMap(x => AstTools.findAsts(x, "FunctionCall")).collect({
+      case y if y.getAttribute("name").sourceString.startsWith("write_") => y
+    })
+
+    val evaluatedExpressionMap = writeFunctionAsts map { ast =>
+      val expression = WdlExpression(ast)
+      val value = expression.evaluate(backendCall.lookupFunction(Map.empty), backendCall.engineFunctions)
+      expression.toWdlString.md5SumShort -> value
+    } toMap
+
+    val writeFunctionFiles = evaluatedExpressionMap collect { case (k, v: Success[_]) => k -> v.get } collect { case (k, v: WdlFile) => k -> Seq(v)}
```

Comment:
I did try this:

```scala
evaluatedExpressionMap collect { case (k, v: Success[WdlFile]) => k -> Seq(v.get) }
```

However, I got a type erasure warning from the compiler, which is why I made it two collects.